src/bin/cargo/commands/install.rs

```diff
@@ -121,6 +121,14 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
             )
             .into());
         }
+
+        if let Ok(url) = crate_name.into_url() {
```

Comment:
I'd prefer we make this a bit more specific and match only `https` or `http` `scheme`s in the URL.

Otherwise, this looks good. We don't need to cover 100% of possible git URLs for a helpful suggestion, and `https` is the most common.

Reply:
Thanks @arlosi. That makes sense. I added the check.

Reply:
I think we need to cover `ssh` as it is also widely used.